puffy-jiggle/Harmony
# Colin/save audio feature

#Title: Add functionality to save generated audio

### This PR adds the saveGeneratedAudio function to enable saving audio files to the backend. It includes:
- A POST request to the backend endpoint /api/save-audio.
- Headers for Content-Type and Authorization using JWT from localStorage.
- Request body containing originalUrl and transformedUrl for the audio files.
- Basic error handling with success and failure alerts.


### This functionality is necessary to store generated audio files persistently in the backend, linking the original audio with its transformed version.

### How can this be tested?
1. Use a valid JWT token in localStorage.
2. Ensure the backend server is running on localhost:4040, use: npm run dev
3. Trigger the saveGeneratedAudio function and verify:
   - A success alert appears if the save is successful.
   - A failure alert appears if the save fails, with an error logged in the console.
   - Go to the db and see if a pair of urls were added (original and transformed)

The backend endpoint /api/save-audio must be available, and a valid JWT token is required for authentication.
Go to the dev tools in the browser to remove the jwt token to see if the button disapears
